pkg/registry/core/shoot/storage/admin_kubeconfig.go

```diff
@@ -224,9 +225,30 @@ func findNewestCACertificate(results []*gardencorev1alpha1.GardenerResourceData)
 }
 
 func getClusterCABundle(resourceDataList gardencorev1alpha1helper.GardenerResourceDataList) ([]byte, error) {
-	var caBundle []byte
+	var (
+		allCAs   = resourceDataList.Select(caCertificateSelector.Add(nameCAClusterReq))
+		caBundle []byte
+	)
+
+	// Sort CAs descending based on their issued-at-time label. This ensures that the current CA is always the first so
+	// that the order in the bundle is the same as in the <shoot-name>.ca-cluster secret.
+	sort.Slice(allCAs, func(i, j int) bool {
+		issuedAtTime1, ok1 := allCAs[i].Labels[secretsmanager.LabelKeyIssuedAtTime]
+		issuedAtTime2, ok2 := allCAs[j].Labels[secretsmanager.LabelKeyIssuedAtTime]
+		if !ok1 || !ok2 {
+			return false
```

Comment:
While doing the review with @ialidzhikov and @ary1992 we were wondering if any logging would be useful here (and on line 245) as it would potentially help when the issued at time cannot be retrieved and if the failure to retrieve them would cause any problems.

Reply:
Instead of logging something, I think it's better to return an error and do not even create a kubeconfig (with a potentially wrong CA).
Thanks for the suggestion, please check again!